x-pack/plugin/wildcard/src/main/java/org/elasticsearch/xpack/wildcard/mapper/AutomatonQueryOnBinaryDv.java

```diff
@@ -92,8 +92,11 @@ public String toString(String field) {
 
     @Override
     public boolean equals(Object obj) {
-        AutomatonQueryOnBinaryDv other = (AutomatonQueryOnBinaryDv) obj;
-        return Objects.equals(field, other.field)  && Objects.equals(matchPattern, other.matchPattern);
+        if (obj instanceof AutomatonQueryOnBinaryDv) {
```

Comment:
I tend to stick with something like:
```
if (obj == null || obj.getClass() != getClass()) {
  return false;
}
AutomatonqueryOnBinaryDv other =...
```

That has the advantage of working "properly" on nulls and subclasses. And I do it from muscle memory....